File: Server/src/main/java/com/api/server/foodb/NcbiTaxonomyMap.java
package com.api.server.foodb;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "ncbi_taxonomy_maps")
public class NcbiTaxonomyMap {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @Size(max = 255)
    @Column(name = "TaxonomyName")
    private String taxonomyName;

    @Size(max = 255)
    @Column(name = "Rank")
    private String rank;

    @NotNull
    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @NotNull
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

}

File: DotNetServer/DotNetServer/Entities/Foodb/NcbiTaxonomyMap.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DotNetServer.Entities.Foodb;

[Table("ncbi_taxonomy_maps")]
public partial class NcbiTaxonomyMap
{
    [Key] [Column("id")] public int Id { get; set; }

    [StringLength(255)] public string? TaxonomyName { get; set; }

    [StringLength(255)] public string? Rank { get; set; }

    [Column("created_at", TypeName = "datetime")]
    public DateTime CreatedAt { get; set; }

    [Column("updated_at", TypeName = "datetime")]
    public DateTime UpdatedAt { get; set; }
}